var/spack/repos/builtin/packages/composable-kernel/package.py

```diff
@@ -64,9 +64,14 @@ def cmake_args(self):
         ]
         if "auto" not in self.spec.variants["amdgpu_target"]:
             args.append(self.define_from_variant("AMDGPU_TARGETS", "amdgpu_target"))
+        if "@5.6.1:" in self.spec:
```

Comment:
You should be using `satisfies` here:

```suggestion
        if self.spec.satisfies("@5.6.1:"):
```

Reply:
See docs for checking specs at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints.